### DISCUSSION TITLE
How to resolve fatal error: 'openssl/conf.h' file not found

### DISCUSSION THREAD

**Discussion by narenkhatwani** [openssl/openssl#21413] [Q&A] [answered]
While running a project, I am encountering the error:

#command-line-arguments
In file included from src/master/master.go:4:
In file included from ./../c/dpf.c:3:
./../c/dpf.h:10:10: fatal error: 'openssl/conf.h' file not found
#include <openssl/conf.h>
         ^~~~~~~~~~~~~~~~
1 error generated.

I tried some solutions mentioned by users at online forums like:

Adding the path to openssl1.1 in my .zshrc file
```bash
echo 'export PATH="/opt/homebrew/opt/openssl@1.1/bin:$PATH"' >> ~/.zshrc
```
I also have set the 

```bash
export LDFLAGS="-L/opt/homebrew/opt/openssl@1.1/lib"
export CPPFLAGS="-I/opt/homebrew/opt/openssl@1.1/include"
export PKG_CONFIG_PATH="/opt/homebrew/opt/openssl@1.1/lib/pkgconfig"
```

System Specifications:
MacOS Version 13.3.1

Can someone please help me with this?

**Comment by paulidale** [comment, marked as answer]
Did you build or install OpenSSL?

Several of the header files are generated and this is one.

Until you build OpenSSL, this file won't exist.  Unless you install it, you'll need to override the include path.

**Comment by narenkhatwani** [comment]
I did a brew install openssl and didn’t build it

**Comment by narenkhatwani** [comment]
Also verified the header file exists at the default path for new packages for MacOS but still I get the same error of the header file not being found.

Would you suggest I brew uninstall and build openssl from source?

**Comment by paulidale** [comment]
I'd suggest raising a bug against brew.

**Comment by narenkhatwani** [comment]
Okay, also just wanted to confirm if there are any steps mentioned anywhere to build OpenSSL on ARM macs

**Comment by paulidale** [comment]
Just follow the install instructions and it should work.
